### PR TITLE
Split `MsgUpdateClient` back into `MsgUpdateClient` and `MsgSubmitMisbehaviour`

### LINKED ISSUE #628: `MsgUpdateClient` domain type does not faithfully represent the IBC protocol messages
## Bug Summary

#584 broke the representation of IBC protocol messages, by creating a single domain type, `MsgUpdateClient`, that represents multiple IBC protocol messages (both `MsgUpdateClient` and `MsgSubmitMisbehavior`).

The domain types should faithfully represent the IBC protocol messages, rather than forcing downstream clients to re-parse them to figure out what message they actually got.

This also breaks the use of the `ibc_proto` traits, forcing callers to write code like
```rust
let msg = <MsgUpdateClient as ibc_proto::protobuf::Protobuf<
    ibc_proto::ibc::core::client::v1::MsgUpdateClient,
>>::decode(raw_action_bytes)?;
```
or
```rust
<ibc::core::ics02_client::msgs::update_client::MsgUpdateClient as Protobuf<
    ibc_proto::ibc::core::client::v1::MsgUpdateClient,
>>::encode_vec(&msg)
```
just to encode or decode messages.

### PR DESCRIPTION
**Pull request by plafer**
Closes: #628

cc @hdevalence

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).
- [x] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/core/ics26_routing/msgs.rs [hunk 1]
```diff
@@ -36,7 +38,7 @@ impl TryFrom<Any> for MsgEnvelope {
                     .map_err(RouterError::MalformedMessageBytes)?;
                 Ok(MsgEnvelope::Client(ClientMsg::CreateClient(domain_msg)))
             }
-            update_client::UPDATE_CLIENT_TYPE_URL => {
+            update_client::TYPE_URL => {
                 let domain_msg =
                     <update_client::MsgUpdateClient as Protobuf<RawMsgUpdateClient>>::decode_vec(
                         &any_msg.value,
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics26_routing/msgs.rs]: Can we write `update_client::MsgUpdateClient::decode_vec(&any_msg.value)` instead?

File: crates/ibc/src/core/ics26_routing/msgs.rs [hunk 2]
```diff
@@ -49,12 +51,12 @@ impl TryFrom<Any> for MsgEnvelope {
                     .map_err(RouterError::MalformedMessageBytes)?;
                 Ok(MsgEnvelope::Client(ClientMsg::UpgradeClient(domain_msg)))
             }
-            update_client::MISBEHAVIOUR_TYPE_URL => {
-                let domain_msg = <update_client::MsgUpdateClient as Protobuf<
+            misbehaviour::TYPE_URL => {
+                let domain_msg = <misbehaviour::MsgSubmitMisbehaviour as Protobuf<
                     RawMsgSubmitMisbehaviour,
                 >>::decode_vec(&any_msg.value)
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics26_routing/msgs.rs]: Same here `misbehaviour::MsgSubmitMisbehaviour::decode_vec(&any_msg.value)` ?

File: crates/ibc/src/core/ics02_client/msgs/update_client.rs [hunk 3]
```diff
@@ -34,8 +19,7 @@ pub enum UpdateKind {
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct MsgUpdateClient {
     pub client_id: ClientId,
-    pub client_message: Any,
-    pub update_kind: UpdateKind,
+    pub header: Any,
     pub signer: Signer,
 }
```
Comment (Farhad-Shabani) [inline crates/ibc/src/core/ics02_client/msgs/update_client.rs:26]: Should we re-add `Msg` implementation to be consistent with others?

```rust
impl Msg for MsgUpdateClient {
    type Raw = RawMsgUpdateClient;

    fn type_url(&self) -> String {
        TYPE_URL.to_string()
    }
}
```

File: crates/ibc/src/core/ics26_routing/msgs.rs (hunk 1 above)
Comment (plafer) [reply crates/ibc/src/core/ics26_routing/msgs.rs]: [60ff485](https://github.com/cosmos/ibc-rs/pull/643/commits/60ff485ee374d9be61e75fcf249c5f6a1f0eab29)

File: crates/ibc/src/core/ics26_routing/msgs.rs (hunk 2 above)
Comment (plafer) [reply crates/ibc/src/core/ics26_routing/msgs.rs]: [60ff485](https://github.com/cosmos/ibc-rs/pull/643/commits/60ff485ee374d9be61e75fcf249c5f6a1f0eab29)

File: crates/ibc/src/core/ics02_client/msgs/update_client.rs (hunk 3 above)
Comment (plafer) [reply crates/ibc/src/core/ics02_client/msgs/update_client.rs:26]: [20f45dd](https://github.com/cosmos/ibc-rs/pull/643/commits/20f45dd969c74a4b383f87f28d5ab8650dd830d3)

**Review by Farhad-Shabani — APPROVED** [review]

**Pull request merged**